integration/qbft/tests/scenario_test.go

```diff
@@ -0,0 +1,194 @@
+package tests
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	spec "github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	spectestingutils "github.com/bloxapp/ssv-spec/types/testingutils"
+	qbftstorage "github.com/bloxapp/ssv/ibft/storage"
+	"github.com/bloxapp/ssv/network"
+	"github.com/bloxapp/ssv/operator/validator"
+	protocolforks "github.com/bloxapp/ssv/protocol/forks"
+	protocolbeacon "github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	protocolstorage "github.com/bloxapp/ssv/protocol/v2/qbft/storage"
+	protocolvalidator "github.com/bloxapp/ssv/protocol/v2/ssv/validator"
+	"github.com/bloxapp/ssv/protocol/v2/types"
+	"github.com/bloxapp/ssv/storage"
+	"github.com/bloxapp/ssv/storage/basedb"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"testing"
+	"time"
+)
+
+var (
+	KeySet4Committee  = spectestingutils.Testing4SharesSet()
+	KeySet7Committee  = spectestingutils.Testing7SharesSet()
+	KeySet10Committee = spectestingutils.Testing10SharesSet()
+	KeySet13Committee = spectestingutils.Testing13SharesSet()
+)
+
+type Scenario struct {
+	Committee           int
+	ExpectedHeight      int
+	Duties              map[spectypes.OperatorID]DutyProperties
+	ValidationFunctions map[spectypes.OperatorID]func(t *testing.T, committee int, actual *protocolstorage.StoredInstance)
+	shared              SharedData
+	validators          map[spectypes.OperatorID]*protocolvalidator.Validator
+}
+
+func (s *Scenario) Run(t *testing.T, role spectypes.BeaconRole) {
+	t.Run(role.String(), func(t *testing.T) {
+		//preparing resources
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		s.validators = map[spectypes.OperatorID]*protocolvalidator.Validator{} //initiating map
+
+		s.shared = GetSharedData(t)
+
+		//initiating validators
+		for id := 1; id <= s.Committee; id++ {
+			id := spectypes.OperatorID(id)
+			s.validators[id] = createValidator(t, ctx, id, getKeySet(s.Committee), s.shared.Logger, s.shared.Nodes[id])
+		}
+
+		//invoking duties
+		for id, dutyProp := range s.Duties {
+			go func(id spectypes.OperatorID, dutyProp DutyProperties) { //launching goroutine for every validator
+				time.Sleep(dutyProp.Delay)
+
+				duty := createDuty(getKeySet(s.Committee).ValidatorPK.Serialize(), dutyProp.Slot, dutyProp.ValidatorIndex, role)
+				require.NoError(t, s.validators[id].StartDuty(duty))
+			}(id, dutyProp)
```

Comment:
why not add to queue? in order to mock real behaviour

Reply:
@egorka-sereda @aaron-blox this is resolved?

Reply:
is resolved